iree/compiler/Dialect/IREE/IR/IREEOps.td

```diff
@@ -167,14 +167,16 @@ def IREE_UnreachableOp : IREE_Op<"unreachable", [NoSideEffect, Terminator]> {
       cond_br %true, ^bb2, ^bb1
     ^bb1:
       // Indicates that this branch should never be taken.
-      iree.unreachable
+      iree.unreachable "shouldn't be here"
     ^bb2:
       ...
 
     ```
   }];
 
-  let assemblyFormat = "attr-dict";
+  let arguments = (ins StrAttr:$message);
```

Comment:
Should it be optional?

Reply:
`llvm_unreachable` requires a message so I figured it'd be best to be consistent here :)